### PR TITLE
Revert "Add caller stacktrace to rethrown RuntimeException (#17249)"

### PR DESCRIPTION
**Pull request by mmedenjak**
This reverts commit 318b336b. Reasoning is that it introduces
significant changes in the invocation handling and causes some tests to
start failing (see https://github.com/hazelcast/hazelcast/issues/6383). 
An alternate or augmented solution must be considered.

EDIT: had to revert https://github.com/hazelcast/hazelcast/pull/17438 as well since it relied on a method in `ExceptionUtil` which was introduced by the above commit.

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2301/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2301/Hazelcast-pr-builder-2301.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] COMPILATION ERROR : 
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/memory/impl/BaseMemoryAccessorTest.java:[28,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:testCompile (default-testCompile) on project hazelcast: Compilation failure: Compilation failure: 
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/memory/impl/BaseMemoryAccessorTest.java:[28,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/memory/impl/BaseMemoryAccessorTest.java:[28,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/memory/impl/BaseMemoryAccessorTest.java:[28,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/memory/impl/BaseMemoryAccessorTest.java:[28,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/memory/impl/BaseMemoryAccessorTest.java:[51,20] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/client/impl/protocol/ClientExceptionFactoryTest.java:[277,42] error: cannot find symbol
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/memory/impl/BaseMemoryAccessorTest.java:[28,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/memory/impl/BaseMemoryAccessorTest.java:[28,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/memory/impl/BaseMemoryAccessorTest.java:[28,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/memory/impl/BaseMemoryAccessorTest.java:[28,15] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/internal/memory/impl/BaseMemoryAccessorTest.java:[51,20] Unsafe is internal proprietary API and may be removed in a future release
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast/src/test/java/com/hazelcast/client/impl/protocol/ClientExceptionFactoryTest.java:[277,42] error: cannot find symbol
--------------------------
</details>
</pre>

**Review by mdogan — APPROVED** [review]

**Review by puzpuzpuz — COMMENTED** [review]
I'm fine with correctness of the revert, but we need to reopen #17202 if we merge this one. And once we reopen it, we have a high prio defect with 4.0.3 milestone. So, we probably need to find a way to fix it.

**Comment by mmedenjak** [comment]
@puzpuzpuz I'm actually doubting the priority of that issue. It says:
> This makes it hard to follow the stack trace. The solution is already available on the member side. This issue is to address the problems on the client-side.

To me it sounds like a usability enhancement, when debugging or analysing issues. And it's not a regression introduced in 4.0.3, seems like it's the consequence of a 4.0 PR (https://github.com/hazelcast/hazelcast/pull/15441).

**Comment by puzpuzpuz** [comment]
@mmedenjak OK, then the issue could be addressed in the next patch release.

**Review by sancar — APPROVED** [review]

**Comment by mmedenjak** [comment]
Thank you for the reviews, everyone! 🙇
